pkg/services/ngalert/api/api_ruler_validation.go

```diff
@@ -189,8 +190,19 @@ func validateRuleGroup(
 			}
 			uids[rule.UID] = idx
 		}
+
+		v := ngmodels.AlertRuleWithOptionals{AlertRule: *rule}
```

Comment:
Please feel free to change the names of these variables to something that might make more sense. I chose these names just to demonstrate how I thought we could solve the issue of propagating optional booleans.